azure_speech_to_text.py

```diff
@@ -145,12 +145,26 @@ def handle_final_result(evt):
         result_future.get()  # wait for voidfuture, so we know engine initialization is done.
         print('Continuous Speech Recognition is now running, say something.')
 
+        request_stop = False
+        timeout_time = time.time()
         while not done:
             # METHOD 1 - Press the stop key. This is 'p' by default but user can provide different key
-            if keyboard.read_key() == stop_key:
-                print("\nEnding azure speech recognition\n")
-                self.azure_speechrecognizer.stop_continuous_recognition_async()
-                break
+            if request_stop == False:
+                if keyboard.read_key() == stop_key:
+                    request_stop = True
+                    timeout_time = time.time()
+                    print("\nAttempting to end azure speech recognition\n")
+            if request_stop == True:
```

Comment:
`if request_stop:` is the right way to do this.

(yes, this works, but == True is unnecessary because python already assumes that request_stop is True when doing `if request_stop`)

Reply:
*this in "yes, this works" refers to the PR code